src/main/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetails.java

```diff
@@ -13,19 +13,27 @@
     @Nullable
     private Integer recommendedLength;
 
+    private boolean allowRichText;
+
     public FeedbackTextQuestionDetails() {
         super(FeedbackQuestionType.TEXT);
         recommendedLength = null;
+        allowRichText = true;
     }
 
     public FeedbackTextQuestionDetails(String questionText) {
         super(FeedbackQuestionType.TEXT, questionText);
         recommendedLength = null;
+        allowRichText = true;
     }
 
     @Override
     public boolean shouldChangesRequireResponseDeletion(FeedbackQuestionDetails newDetails) {
-        return false;
+        assert newDetails instanceof FeedbackTextQuestionDetails;
+
+        // delete the existing response upon change from rich text allowed to disallowed
+        // due to the effort to cleanup of HTML tags from the respondents
+        return !((FeedbackTextQuestionDetails) newDetails).allowRichText && allowRichText;
```

Comment:
Take a look at how `shouldChangesRequireResponseDeletion` is called. Do you need cast operation here?

Reply:
https://github.com/TEAMMATES/teammates/blob/60400d67c305bfa76cfadcd16c49bb353f532224/src/main/java/teammates/common/datatransfer/questions/FeedbackContributionQuestionDetails.java#L40-L41

I refer to the above lines, where there is an explicit casting to get the ContributionQuestionDetails specific attributes. I assumed doing it inline like the above would not raise any eyebrows.

Reply:
And that this part needs to be extracted out, as the behaviour changes according to the exact values this attribute holds.

Reply:
My bad. It's fine then :D